lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8437,6 +8438,107 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
   }
 }
 
+namespace {
+class PointerParamInfo {
+public:
+  virtual void print(clang::ASTContext &ctx, llvm::raw_ostream &out) const = 0;
+  virtual ~PointerParamInfo() {}
+};
+
+class CountedByParam : public PointerParamInfo {
+public:
+  size_t pointerIndex;
+  clang::Expr *countExpr;
+  bool isSizedBy;
+  CountedByParam(size_t idx, clang::Expr *E, bool sizedBy)
+      : pointerIndex(idx), countExpr(E), isSizedBy(sizedBy) {}
+
+  virtual void print(clang::ASTContext &ctx,
+                     llvm::raw_ostream &out) const override {
+    out << ".";
+    if (isSizedBy)
+      out << "sizedBy";
+    else
+      out << "countedBy";
+    out << "(pointer: " << pointerIndex << ", ";
+    if (isSizedBy)
+      out << "size";
+    else
+      out << "count";
+    out << ": \"";
+    countExpr->printPretty(
+        out, {}, {ctx.getLangOpts()}); // TODO: map clang::Expr to Swift Expr
+    out << "\")";
+  }
+  virtual ~CountedByParam() {}
+};
+} // namespace
+
+void ClangImporter::Implementation::importBoundsAttributes(
+    FuncDecl *MappedDecl) {
+  auto ClangDecl =
+      dyn_cast_or_null<clang::FunctionDecl>(MappedDecl->getClangDecl());
+  if (!ClangDecl)
```

Comment:
Could this be an assert? Do we ever expect this function to be invoked on something that does not have a clang declaration or the clang declaration is not a function declaration?

Reply:
I found this comment in another function:
```
  // sourceDecl->getClangDecl() can be null because some lazily instantiated cases like C++ members that were instantiated from using-shadow-decls have no corresponding Clang decl.
```
My main motivation was that it is how `importSwiftAttrAttributes()` does it however, and the same argument is passed to both functions.